chrome/browser/ui/webui/record_replay/record_replay_ui.cc

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "base/memory/ref_counted_memory.h"
+#include "chrome/browser/ui/webui/record_replay/record_replay_ui.h"
+#include "chrome/common/webui_url_constants.h"
+#include "chrome/grit/record_replay_resources.h"
+#include "chrome/grit/record_replay_resources_map.h"
+#include "content/public/browser/web_contents.h"
+
+RecordReplayUI::RecordReplayUI(
+    content::WebUI* web_ui)
+    : ui::MojoWebUIController(web_ui) {
+  content::WebUIDataSource* data_source =
+      content::WebUIDataSource::CreateAndAdd(
+          web_ui->GetWebContents()->GetBrowserContext(),
+          chrome::kChromeUIRecordReplayHost);
+
+  data_source->AddResourcePaths(base::make_span(
+      kRecordReplayResources, kRecordReplayResourcesSize));
+  data_source->AddResourcePath("", IDR_RECORD_REPLAY_RECORD_REPLAY_HTML);
```

Comment:
This call registers the empty path, mapping it to our HTML.  I'm not 100% sure this is necessary, but it feels like it is, given the url we're loading is `chrome://recordreplay/`, without a file.